scripts/Dockerfile

```diff
@@ -1,3 +1,61 @@
+# Find the run target below for an example showing how it's possible to install Klipper in Docker.
+
+# But first, here is the docker builder environment target. Use this to build klipper from any
+# platform without installing locally the required packages and compilers. This is especially usefull
+# on old linux distribution and on non-linux OS'es.
+# IMPORTANT: The docker build must be run from the root of the repo. To use docker builder env
+# either prefix your command with ./scripts/with-docker-builder-env.sh like this:
+#       ./scripts/with-docker-builder-env.sh make menuconfig
+# or invoke it manually like so:
+#       docker build . -f scripts/Dockerfile --target builder -t klipper-builder
+#       docker run -it --rm -u `id -u`:`id -g` -v `pwd`:`pwd`:rw -w `pwd` klipper-builder make menuconfig
+FROM ubuntu:20.04 as builder
+ENV DEBIAN_FRONTEND=noninteractive
+# TZ hack to bypass timezone selection prompt
+ENV TZ=GMT
+RUN ln -snf "/usr/share/zoneinfo/$TZ" /etc/localtime
+RUN echo "$TZ" > /etc/timezone
+# Install klipper build deps
+RUN apt-get update && apt-get install -y --no-install-recommends \
+        avr-libc \
+        binutils-arm-none-eabi \
+        bison \
+        build-essential \
+        flex \
+        gcc-arm-none-eabi \
+        gcc-avr \
+        libffi-dev \
+        libgmp-dev \
+        libmpc-dev \
+        libmpfr-dev \
+        libnewlib-arm-none-eabi \
+        pv \
+        python2-dev \
```

Comment:
python2?

Reply:
> I would much prefer see this move to python3 instead

What is needed for that?

Reply:
Happy to look into that. Will give that a try.